astroid/interpreter/objectmodel.py

```diff
@@ -119,7 +119,35 @@ def lookup(self, name):
         raise AttributeInferenceError(target=self._instance, attribute=name)
 
 
-class ModuleModel(ObjectModel):
+class DunderCompletionMixin:
```

Comment:
How about making a list of what we consider public API, something as small as possible. ``Run``, ``run_pylint``, ``BaseChecker``, ``BaseReporter``, and nothing else until someone ask for it worriedly ? We're still going to have to be more careful under the hood but at least it will remove the uncertainty.